packages/models/files.js

```diff
@@ -7,6 +7,12 @@ const Files = new FilesCollection({
     debug: true,
     collectionName: 'files',
     protected(fileObj) {
+        var session = this.params.query.xmtok;
+        var userId  = (Meteor.server.sessions[session] && Meteor.server.sessions[session].userId) ? Meteor.server.sessions[session].userId : null;
```

Comment:
@dr-dimitru @menelike how about just passing both, the `userId` and the `loginToken` from the client. Then hash the token und fetch the user using:

```
const user = Meteor.users.find({
    _id: userId,
    'services.resume.loginTokens.hashedToken': hashToken,
});
```
?